Kitodo/src/main/java/org/kitodo/production/interfaces/activemq/TaskActionProcessor.java

```diff
@@ -0,0 +1,219 @@
+                break;
+            case ERROR_CLOSE:
+                if ((!mapMessageObjectReader.hasField(KEY_CORRECTION_TASK_ID) && !TaskStatus.INWORK.equals(
+                        currentTask.getProcessingStatus())) || (mapMessageObjectReader.hasField(
+                        KEY_CORRECTION_TASK_ID) && !TaskStatus.LOCKED.equals(currentTask.getProcessingStatus()))) {
+                    throw new ProcessorException(
+                            "Status of task is not INWORK if there is a no corrected task ID or LOCKED if there is a corrected task ID.");
+                }
+                actionErrorClose(mapMessageObjectReader, currentTask, currentUser);
+                break;
+            case CLOSE:
+                workflowControllerService.closeTaskByUser(currentTask);
+                break;
+            default:
+                if (!mapMessageObjectReader.hasField(KEY_MESSAGE)) {
+                    throw new ProcessorException("Message field of task action COMMENT is required.");
+                }
+        }
+
+        if (Objects.nonNull(comment)) {
+            ServiceManager.getCommentService().saveToDatabase(comment);
+        }
+    }
+
+    private static Comment buildComment(Task currentTask, String message) {
+        Comment comment = new Comment();
+        comment.setProcess(currentTask.getProcess());
+        comment.setAuthor(ServiceManager.getUserService().getCurrentUser());
+        comment.setMessage(message);
+        comment.setCreationDate(new Date());
+        comment.setType(CommentType.INFO);
+        comment.setCurrentTask(currentTask);
+        return comment;
+    }
+
+    private void actionErrorOpen(MapMessageObjectReader mapMessageObjectReader, Comment comment)
+            throws ProcessorException, JMSException, DAOException, DataException {
+        if (mapMessageObjectReader.hasField(KEY_CORRECTION_TASK_ID)) {
+            Integer correctionTaskId = mapMessageObjectReader.getMandatoryInteger(KEY_CORRECTION_TASK_ID);
+            Task correctionTask = taskService.getById(correctionTaskId);
+            if (Objects.isNull(correctionTask)) {
+                throw new ProcessorException("Correction task with id " + correctionTaskId + " not found.");
+            }
+            comment.setCorrectionTask(correctionTask);
+        }
+        comment.setType(CommentType.ERROR);
+        workflowControllerService.reportProblem(comment, TaskEditType.QUEUE);
+    }
+
+    private void actionProcess(Task currentTask, User currentUser) throws DataException {
+        currentTask.setProcessingStatus(TaskStatus.INWORK);
+        currentTask.setEditType(TaskEditType.QUEUE);
+        currentTask.setProcessingTime(new Date());
+        taskService.replaceProcessingUser(currentTask, currentUser);
+        if (Objects.isNull(currentTask.getProcessingBegin())) {
+            currentTask.setProcessingBegin(new Date());
+            taskService.save(currentTask);
+        }
```

Comment:
Task get only saved on the first time ever. If this task is executed a second time, the change is only submitted to the index and not to the database which is resulting in different task state between database and index. I would recommend to set the processing begin every time the task is set to "in process" and save the task even every time.